model/span/event.go

```diff
@@ -447,5 +449,46 @@ func (e *Event) fields(tctx *transform.Context) common.MapStr {
 
 	st := e.Stacktrace.Transform(tctx)
 	utility.Set(fields, "stacktrace", st)
+
+	if h := e.servicemapHash(tctx); h != "" {
+		utility.Set(fields, "servicemap", common.MapStr{"fingerprint": h})
+	}
 	return fields
 }
+
+func (e *Event) servicemapHash(tctx *transform.Context) string {
+	if e.Destination == nil || e.Destination.Address == nil || *e.Destination.Address == "" {
+		return ""
+	}
+	h := xxhash.New()
+	fmt.Fprintf(h, "%s|%s", *e.Destination.Address, e.Type)
+	if e.Subtype != nil {
+		fmt.Fprintf(h, "|%s", *e.Subtype)
+	}
+	if name := serviceVal("name", e.Service, tctx.Metadata.Service); name != "" {
+		fmt.Fprintf(h, "|%s", name)
+	}
+	if env := serviceVal("env", e.Service, tctx.Metadata.Service); env != "" {
+		fmt.Fprintf(h, "|%s", env)
+	}
+
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func serviceVal(key string, services ...*metadata.Service) string {
```

Comment:
per discussion, split into two functions rather than using the key.  Looking forward to results of whether return should be `string` or `*string`